specification/resources/resource-manager/Microsoft.Authorization/preview/2022-08-01-preview/policyVariableValues.json

```diff
@@ -0,0 +1,548 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/VariableValue"
+            },
+            "description": "Parameters for the variable value."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the updated variable value.",
+            "schema": {
+              "$ref": "#/definitions/VariableValue"
+            }
+          },
+          "201": {
+            "description": "Created - Returns information about the new variable value.",
+            "schema": {
+              "$ref": "#/definitions/VariableValue"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "VariableValues"
+        ],
+        "operationId": "VariableValues_GetAtManagementGroup",
+        "summary": "Retrieves a variable value.",
+        "description": "This operation retrieves a single variable value; given its name,  management group it was created at and the variable it's created for.",
+        "x-ms-examples": {
+          "Retrieve a variable value at management group": {
+            "$ref": "./examples/getVariableValueAtManagementGroup.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ManagementGroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/VariableNameParameter"
+          },
+          {
+            "$ref": "#/parameters/VariableValueNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the variable value.",
+            "schema": {
+              "$ref": "#/definitions/VariableValue"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudError": {
+      "x-ms-external": true,
+      "properties": {
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+        }
+      },
+      "type": "object",
+      "description": "An error response from a policy operation."
+    },
+    "PolicyVariableValueColumnValue": {
+      "required": [
+        "columnName",
+        "columnValue"
+      ],
+      "properties": {
+        "columnName": {
+          "type": "string",
+          "description": "Column name for the variable value"
+        },
+        "columnValue": {
```

Comment:
@leni-msft , I want this columnValue to be a JValue, i.e. it can be an int, or a string etc. What is the best way to represent this in swagger?

Reply:
From swagger perspective, you can enumerate all the supported types in `type` as it could be an array, like `"type": [ "integer", "number", "boolean", "null", "string" ]` in your case.

However not sure if SDKs can generate code succesfully. @weidongxu-microsoft @ArcturusZhang @msyyc Any suggestion for multi types property?

Reply:
I don't think swagger 2.0 have support for multiple types. And `JValue` is .NET concept, which likely not having matching concept in other languages.

In this case, you might had to go with `type: any`?

Reply:
How does "any" get generated in Java do you know? How about enumerating all the possible values in a sub type or something like that, would that be less likely to break the sdk generation?

Reply:
I don't think autorest likes "any" 

![image](https://user-images.githubusercontent.com/1864429/184242611-b07fb792-a749-43ea-8d71-b09b80bd62aa.png)

Reply:
Java will have `Object` to both `object` and `any`.

However it could mean different thing for other languages, as `any` means anything including primitives like int, float; and `object` only means any JSON object (the dict but not primitives).
@tadelesh for advice on Go.

---

Sorry, to be precise, `type: any` mean does not write the `type` in the swagger definition for this property.

Reply:
Go now generate same for both (major reason is swagger is usually wrong).

Python recommend to use `any`, but they will work for the `object`.

Reply:
thanks @weidongxu-microsoft.

@cemheren I see in your swagger examples, both `string` and `integer` values have been there, and it passed the model validation. So you can keep using `type: object`.

Reply:
Thanks @leni-msft and @weidongxu-microsoft. I'm not sure if I messed up the syntax for type: any, I can try that if you can show me what to use. (I don't think schema validation is helping me on this either.) 

Otherwise are we ok with me closing this comment? Seems like object ~roughly~ works for this scenario. Am I right?

Reply:
Sure. I am fine with "object" (given it is already used extensively in policy https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Authorization/stable/2021-06-01/policyDefinitions.json#L564-L575).